R/mock.R

```diff
@@ -81,11 +85,11 @@ mock_this_exercise <- function(
   .label = "mock",
   .engine = "r",
   .stage = "check",
+  .result = rlang::missing_arg(),
```

Comment:
What's the difference between `.result = rlang::missing_arg()` and just `.result` with no default value?

Reply:
If `.result` isn't provided, we want to create `.result` from the `.user_code`, provided that it's R code that we can evaluate.

If we don't have R code though, we need the user to tell us what the result will be. That result _might be_ `NULL`, so I'm using `rlang::missing_arg()` to tell the difference.

We could leave `.result` without a default value and use `missing(.result)` to catch that, but I prefer being explicit. I also prefer using "no default value" as a strong "required argument" signal; in this case `.result` isn't required.